### PR TITLE
WIP: recursive convolution for Gaussian smoothing and derivative kernels

### PR DESCRIPTION
**Pull request by svenpeter42**

### CODE REVIEW

File: include/vigra/recursive_multi_convolution.hxx [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+        std::pair<Shape, Shape> res;
+        res.first = from_point;
+        res.second = to_point;
+        return res;
+    }
+};
+
+#ifndef M_SQRT2PI
+/** \brief The constant sqrt(2 pi) */
+#define M_SQRT2PI 2.50662827463100050241576528481104525
```
Comment (ukoethe) [inline include/vigra/recursive_multi_convolution.hxx]: This belongs into mathutil.hxx

File: include/vigra/recursive_multi_convolution.hxx [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+        return *this;                                                          \
+    }                                                                          \
+    typename ParamVec::p_vector get##getter_setter_name() const {              \
+        return member_name.vec;                                                \
+    }                                                                          \
+    void set##getter_setter_name(typename ParamVec::p_vector vec) {            \
+        member_name.vec = vec;                                                 \
+    }
+
+template <unsigned dim> class RecursiveConvolutionOptions {
```
Comment (ukoethe) [inline include/vigra/recursive_multi_convolution.hxx]: I think you can just add the function `recursiveOrder()` to the existing `ConvolutionOptions` class

**Comment by jakirkham** [comment]
May want to merge/rebase on `version-1-11-rc`. It includes some needed CI improvements.

**Comment by svenpeter42** [comment]
I will wait until that branch is merged into master. I don't think this will go into 1-11-rc and rebasing now and then still trying to merge into master would likely create a huge mess.

**Comment by svenpeter42** [comment]
going to create a new one because i rebased on the next release branch.

**Comment by svenpeter42** [comment]
https://github.com/ukoethe/vigra/pull/341

**Pull request closed without merging**